airbyte-integrations/connectors/source-notion/source_notion/source.py

```diff
@@ -41,7 +41,18 @@ def check_connection(self, logger: AirbyteLogger, config: Mapping[str, Any]) ->
             next(records)
             return True, None
         except requests.exceptions.RequestException as e:
-            return False, e
+            # The most likely user error will be incorrectly configured credentials. We can provide a specific error message for those cases. Otherwise, the stock Notion API message should suffice.
+            if e.response.json()["code"] == "unauthorized":
+                return (
+                    False,
+                    "The provided API access token is invalid. Please double-check that you input the correct token and have granted the necessary permissions to your Notion integration.",
+                )
+            if e.response.json()["code"] == "restricted_resource":
+                return (
+                    False,
+                    "The provided API access token does not have the correct permissions configured. Please double-check that you have granted all the necessary permissions to your Notion integration.",
+                )
+            return False, f"{e.response.json()['message']}"
```

Comment:
better to use `.get('message', "<default error message>")` in case the response from notion didn't have a `message` field